### PR TITLE
Prevent Synthetic's from removing their organs to be immune to EMP effects with no drawbacks

### PR DESCRIPTION
**Pull request by nevimer**
## About The Pull Request

Right, so the species have a ton of flags that means they don't really need internal organs to begin with - they only feel the effects when they're dead organs or EMP'd organs - and it's only stunlocking. This means you can gut yourself with  no drawbacks, so why wouldn't you? Plus, this prevents you from giving your OP lungs to organics to circumvent their OWN need to breathe. 

Also makes it so revive for putting a brain into and IPC is 50 or less damage, instead of 49 or less. 
dreamseeker_6LljHiLOwa

## How This Contributes To The Skyrat Roleplay Experience
Fixes a possible abuse case. 

## Changelog

:cl:
fix: Fixes Synthetics being table to remove their organs for all gain. You can still remove head organs and the brain. 
:/cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/surgery/organs/ipc.dm [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	if(user.stat == DEAD && ishuman(user))
 		var/mob/living/carbon/human/user_human = user
 		if(user_human?.dna?.species && (REVIVES_BY_HEALING in user_human.dna.species.species_traits))
-			if(user_human.health > 50)
+			if(user_human.health >= 50)
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/surgery/organs/ipc.dm:20]: unmentioned change

Comment (nevimer) [reply modular_skyrat/modules/customization/modules/surgery/organs/ipc.dm:20]: mentioned now!

**Comment by tf-4** [comment]
Your changelog is broken

**Comment by nevimer** [comment]
> Your changelog is broken

based thanks

**Comment by Stonetear** [comment]
I really, really dislike this since this means you can no longer do any kind of cool surgeries with exotic organs on synths.  Synths already cannot be upgraded with the advanced surgeries, can't get beneficial diseases, and can't use healing chems or healing gasses.
This really removes the only remaining medical content for synths to really do.

If we're going to do this, why not make it so that Plasmamen can't remove _their_ organs (since their organs are a EVEN MORE of a direct upgrade!)  Why not make it so Etherials can't have their organs removed (unfair advantage, of course), too.  This is like the edge case of people picking nerve stapled and having robotics hack their head off to remove it.

**Comment by YakumoChen** [comment]
The actual solution is to tie the species traits to the organs themselves and not give a bunch of strong buffs to the species for existing.

Then make the organs incompatible with non-synth species (make it check for blood type or just species, whichever is saner)

This also future proof existing problems with synths trading their organs in at character selection and realising that it fucks them over.

As far as the other arguments go, it would make sense to tie, say, plasmaman organ functionality to the plasma breathing requirement, or make ethereal organs require a working ethereal heart, but that sounds like an upstream problem.

**Comment by nevimer** [comment]
> The actual solution is to tie the species traits to the organs themselves and not give a bunch of strong buffs to the species for existing.
> 
> Then make the organs incompatible with non-synth species (make it check for blood type or just species, whichever is saner)
> 
> This also future proof existing problems with synths trading their organs in at character selection and realising that it fucks them over.
> 
> As far as the other arguments go, it would make sense to tie, say, plasmaman organ functionality to the plasma breathing requirement, or make ethereal organs require a working ethereal heart, but that sounds like an upstream problem.

The organs are null organs. They don't do anything, at all. It's the species flags that make them immune to everything, so  when the parent organ code runs, it just hits return. The only beneficial organ they have to transplant is the lung, and it doesn't need to exist because despite it being 0'd out, they have nobreathe anyways as a species trait. 

> I really, really dislike this since this means you can no longer do any kind of cool surgeries with exotic organs on synths. Synths already cannot be upgraded with the advanced surgeries, can't get beneficial diseases, and can't use healing chems or healing gasses. This really removes the only remaining medical content for synths to really do.
> 
> If we're going to do this, why not make it so that Plasmamen can't remove _their_ organs (since their organs are a EVEN MORE of a direct upgrade!) Why not make it so Etherials can't have their organs removed (unfair advantage, of course), too. This is like the edge case of people picking nerve stapled and having robotics hack their head off to remove it.

This is part one in a series of pull requests. You have no reason to remove them as it breaks the game. You need to understand right now these organs have no code. They all just hit early returns (species traits), and are otherwise normal-ass basic cybernetics.

**Comment by nevimer** [comment]
Even if you DID put new organs in, the code for them would never run, at all, ever, hence breaking the game and having no reason to do it.

**Comment by YakumoChen** [comment]
> The organs are null organs. They don't do anything, at all. It's the species flags that make them immune to everything, so when the parent organ code runs, it just hits return. The only beneficial organ they have to transplant is the lung, and it doesn't need to exist because despite it being 0'd out, they have nobreathe anyways as a species trait.

Yes, move some species flags to the organs, make the organs require being a synth in some way to function correctly. Make the organs require being present for the synth to function so that they can't just be removed for no reason. Making the organs unremovable opens up a different set of problems that you won't have foreseen. 

At present I can already tell this will affect abductors, who can remove synth hearts and insert their own manually for their experiments.

**Comment by nevimer** [comment]
> > The organs are null organs. They don't do anything, at all. It's the species flags that make them immune to everything, so when the parent organ code runs, it just hits return. The only beneficial organ they have to transplant is the lung, and it doesn't need to exist because despite it being 0'd out, they have nobreathe anyways as a species trait.
> 
> Yes, move some species flags to the organs, make the organs require being a synth in some way to function correctly. Make the organs require being present for the synth to function so that they can't just be removed for no reason. Making the organs unremovable opens up a different set of problems that you won't have foreseen.
> 
> At present I can already tell this will affect abductors, who can remove synth hearts and insert their own manually for their experiments.

The Abductor surgery has failsafes for not finding an applicable organ.

**Comment by YakumoChen** [comment]
You can't insert an abductor heart if there is already a heart in the chest.

It is also not "the abductor surgery". Synth abduction must make use of Manipulate Organs to properly probe a synth.

As before, making organs unremovable will only lead to further problems down the line. It's better to just make synths more dependent on their organs, and to code means to prevent other races from using them. This also makes organ failure more meaningful since it could actually kill the synth.

**Comment by nevimer** [comment]
> You can't insert an abductor heart if there is already a heart in the chest.
> 
> It is also not "the abductor surgery". Synth abduction must make use of Manipulate Organs to properly probe a synth.
> 
> As before, making organs unremovable will only lead to further problems down the line. It's better to just make synths more dependent on their organs, and to code means to prevent other races from using them. This also makes organ failure more meaningful since it could actually kill the synth.

Again, as stated before, this is one PR in a series of PR's. Currently removing any organs breaks the synth completely.

**Comment by YakumoChen** [comment]
What do you plan further in the "series"?

Even knowing that I'm against this change. A temporary fix is often the most permanent solution. This breaks more than it fixes and therefore is a bad solution.

Players who are knowingly exploiting things by having their organs removed open themselves up to bwoinks so I don't believe in this change until better workarounds (such as my own suggestions) are properly implemented.

**Comment by nevimer** [comment]
In their current state, a synthetic has everything to gain by being gutted. This fixes that, and that is the scope of the PR. When the time comes, I will reintroduce the ability to remove their organs. But currently? This is not the time.

**Comment by nevimer** [comment]
> What do you plan further in the "series"?
> 
> Even knowing that I'm against this change. A temporary fix is often the most permanent solution. This breaks more than it fixes and therefore is a bad solution.
> 
> Players who are knowingly exploiting things by having their organs removed open themselves up to bwoinks so I don't believe in this change until better workarounds (such as my own suggestions) are properly implemented.


Synth:

https://user-images.githubusercontent.com/77420409/179124946-b135bd98-44ab-4811-815a-152d2ca8907f.mp4



Human:

https://user-images.githubusercontent.com/77420409/179124898-c191dea0-ce05-48bd-b266-fd506235260e.mp4

**Comment by YakumoChen** [comment]
> In their current state, a synthetic has everything to gain by being gutted. This fixes that, and that is the scope of the PR. When the time comes, I will reintroduce the ability to remove their organs. But currently? This is not the time.

when is this ever relevant in-game? you can't disembowel synths anyway, they're immune to wounding. The only way to remove their organs is to do so manually.

This PR is simply fixing a non-issue by creating a different problem, instead of actually fixing the issue.

**Pull request closed without merging**